plate_mapper/plate_mapper.py

```diff
@@ -153,8 +155,12 @@ def plate_mapper(input_f, barseq_f, output_f, names_f=None, special_f=None):
                 # empty well (if defined as a special sample)
                 metadatum = specs['']['metadatum']
                 sample = '%s%s.%s' % (specs['']['name'], pid, well)
-        output_f.write('%s\t%s\n' % ('\t'.join((sample, barcode, primer, plate,
-                                     well)), '\t'.join(metadatum)))
+        if sample or empty:
+            # replace underscore and dash with dot in sample name
+            sample_dot = sample.replace('_', '.').replace('-', '.')
+            output_f.write('%s\t%s\n' % (
+                '\t'.join((sample_dot, barcode, primer, plate, well)),
```

Comment:
I don't think that will work. The former `join` has two sets of parentheses, meaning that `sample_dot`, `barcode`, etc. are scalars. But the latter `join` directly operates on a list (`metadatum`).